src/Sulu/Bundle/MediaBundle/Media/Manager/DefaultMediaManager.php

```diff
@@ -822,9 +826,32 @@ public function addFormatsAndUrl(Media $media)
             )
         );
 
-        $media->setUrl(
-            $this->getUrl($media->getId(), $media->getName(), $media->getVersion())
-        );
+        // Set Version Urls
+        $versionData = array();
+        foreach ($media->getFile()->getFileVersions() as $fileVersion) {
+            $versionData[$fileVersion->getVersion()] = array();
+            $versionData[$fileVersion->getVersion()]['url'] = $this->getUrl(
+                $media->getId(),
+                $fileVersion->getName(),
+                $fileVersion->getVersion()
+            );
+        }
+
+        $media->setAdditionalVersionData($versionData);
+
+        // Set Current Url
+        if (
+            isset($versionUrls[$media->getVersion()])
+            && isset($versionUrls[$media->getVersion()]['url'])
+        ) {
+            $media->setUrl($versionUrls[$media->getVersion()]['url']);
+        } else {
```

Comment:
`else` is not required here (it should [generally be avoided](http://williamdurand.fr/2013/06/03/object-calisthenics/#2-don-t-use-the-else-keyword)) as you can return `$media` on the previous line